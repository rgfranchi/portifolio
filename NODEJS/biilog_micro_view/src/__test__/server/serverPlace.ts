import { Response } from "miragejs";
import { IFormSelectValue } from "../../components/FormsComponent";
import { IPlace } from "../../Place/PlaceData";
const baseURL = "place";

export const placeRoutes = (actions: any) => {
  let headers = {};
  // actions.get(`${baseURL}/findAll`, (schema: any) => {
  //   const dataBase = schema.db.groups;
  //   if (dataBase.length === 0) {
  //     return new Response(204, headers, "");
  //   }
  //   return new Response(200, headers, dataBase);
  // });
  actions.get(
    `${baseURL}/listPlaceSelect/:companyId`,
    (schema: any, request: any) => {
      const company_id = request.params.companyId;
      if (company_id.toString() === "9999") {
        return new Response(404, headers, {
          errors: ["Place Id Not Found"],
        });
      }
      const dataBase = schema.db.places.where(
        (places: IPlace) =>
          places.companyId.toString() === company_id.toString()
      );
      let dataSelect: IFormSelectValue[] = [];
      dataBase.forEach((value: any) => {
        // @ts-ignore
        dataSelect.push({ id: parseInt(value.id), value: value.nome });
      });
      if (dataSelect.length === 0) {
        return new Response(204, headers, "");
      }
      return new Response(200, headers, dataSelect);
    }
  );
};

/**
 * Valores MOCk para teste da entidade.
 */
export const listPlaceMock: IPlace[] = [
  {
    id: 1,
    nome: "BIILOG",
    codPostal: "04557-030",
    pais: "Brasil",
    cidade: "São Paulo",
    bairro: "Brooklin Paulista",
    logradouro: "Rua Marquês de Cascais",
    numero: "152",
    complemento: "Sala 2",
    latitude: -23.60362371,
    longitude: -46.68348267,
    cercaVirtual:
      "[{-23.60294043790386, -46.6841210389477},{-23.60415459596241, -46.682597544328736},{-23.60491159570152, -46.68327346095547},{-23.60367778219504, -46.68484523533348}]",
    observacao: "Informações de teste",
    companyId: 1,
  },
  {
    id: 2,
    nome: "WINE",
    codPostal: "29168-090",
    pais: "Brasil",
    cidade: "Serra",
    bairro: "Civit II",
    logradouro: "Rua Comendador Alcídes Simão Helou",
    numero: "1478",
    complemento: null,
    latitude: -20.18118613,
    longitude: -40.23491962,
    cercaVirtual:
      "[{-20.17906131735574, -40.235627718837044},{-20.179428882535753, -40.231615134418085},{-20.181926288755346, -40.23581010903791},{-20.180632274801525, -40.23177070253059}]",
    observacao: "Informações de teste",
    companyId: 2,
  },
  {
    id: 3,
    nome: "LinconLN",
    codPostal: "07170-350",
    pais: "Brasil",
    cidade: "Guarulhos",
    bairro: "Vila Aeroporto",
    logradouro: "Av. Papa João Paulo I",
    numero: "1818",
    complemento: null,
    latitude: -23.4360454,
    longitude: -46.44681009,
    cercaVirtual:
      "[{-23.433722250373634, -46.4492133459274},{-23.437334923017147, -46.44861253114809},{-23.437482577836786, -46.44419225098603},{-23.433978192284833, -46.44422443749206}]",
    observacao: "Informações de teste",
    companyId: 3,
  },
  {
    id: 4,
    nome: "CNHi",
    codPostal: "18087-220",
    pais: "Brasil",
    cidade: "Sorocaba",
    bairro: "Éden",
    logradouro: "Avenida Jerome Case",
    numero: "1801",
    complemento: null,
    latitude: -23.43962332,
    longitude: -47.39056233,
    cercaVirtual:
      "[{-23.43669977065856, -47.39243987717196},{-23.438068038911084, -47.387933766327144},{-23.44110967801963, -47.38826636022283},{-23.440893123504654, -47.39275101339696}]",
    observacao: "Informações de teste",
    companyId: 4,
  },
];

export let loadPlacesByCompany = (company_id: number): IPlace[] => {
  return listPlaceMock.filter((places) => {
    return places.companyId === company_id;
  });
};
