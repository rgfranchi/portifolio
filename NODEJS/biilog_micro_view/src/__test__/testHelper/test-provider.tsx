import React from "react";
import { MemoryRouter, Route, Router } from "react-router-dom";

import { objMockSession } from "./test-session";

import { Provider } from "react-redux";

//componentes de tela.
import { ModalLoading } from "../../components/ModalsComponent";

/**
 * Renderiza valores para teste.
 * Utilizando o render do framework @testing-library/react
 * Componente ModalLoading.
 * @param {*} render
 * @param {*} ui
 * @param {*} history
 * @param {*} store
 * @returns
 */
export const renderProviderRouter = (
  render: any,
  ui: any,
  history: any,
  store: any,
  modal: boolean = true
) => {
  // @ts-ignore
  function Wrapper({ children }) {
    return providerRouter(children, history, store, modal);
  }
  // @ts-ignore
  return render(ui, { wrapper: Wrapper });
};

/**
 * ConstrÃ³i com base no renderProviderRouter incluindo MemoryRouter e Router.
 * @param {*} render
 * @param {*} ui
 * @param {*} history
 * @param {*} store
 * @param {*} initialEntries
 * @param {*} path
 * @returns
 */
export const renderProviderMemoryRouter = (
  render: any,
  ui: any,
  history: any,
  store: any,
  initialEntries: any,
  path: string,
  modal: boolean = true
) => {
  const wrapper = (
    <MemoryRouter initialEntries={[initialEntries]}>
      <Route path={path}>{ui}</Route>
    </MemoryRouter>
  );
  return renderProviderRouter(render, wrapper, history, store, modal);
};

/**
 * Cria as funcionalidades do Provider e Router
 * Utilizando o mount do framework Enzyme
 * @param {*} mount
 * @param {*} children
 * @param {*} history
 * @param {*} store
 * @param {boolean} modal
 * @returns
 */
export const mountProviderRouter = async (
  mount: any,
  children: any,
  history: any,
  store: any,
  modal: boolean = true
) => {
  return await mount(providerRouter(children, history, store, modal));
};

export const createSessionCurrentUser = (store: any) => {
  store.dispatch({
    type: "session/currentUser",
    payload: objMockSession,
  });
};

const providerRouter = (
  children: any,
  history: any,
  store: any,
  modal: boolean
) => {
  return (
    <Provider store={store}>
      <Router history={history}>
        {modal && <ModalLoading />}
        {children}
      </Router>
    </Provider>
  );
};
