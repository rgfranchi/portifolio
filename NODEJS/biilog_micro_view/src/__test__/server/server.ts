import { createServer, RestSerializer, Model } from "miragejs";
import { userRoutes } from "./serverUser";
import { listCompanyMock, companyRoutes } from "./serverCompany";
import { listOperatorMock, operatorRoutes } from "./serverOperator";
import { listUserMock } from "./serverUser";
import { listGroupMock, groupRoutes } from "./serverGroup";
import { listForkliftMock, forkliftRoutes } from "./serverForklift";
import { listPlaceMock, placeRoutes } from "./serverPlace";

let isEmpty = false;
// variÃ¡vel para resposta vazia, impede alimentar com array original.
/**
 *
 * @param {boolean} empty
 */
export const forceEmpty = (empty: boolean) => {
  // console.info("ForceEmpty", empty);
  isEmpty = empty;
};

export const getIsEmpty = () => isEmpty;

// https://miragejs.com/docs/main-concepts/database/
// https://miragejs.com/api/classes/db-collection/
// https://miragejs.com/docs/main-concepts/route-handlers/
export const mockServer = () => {
  return createServer({
    serializers: {
      reminder: RestSerializer.extend({
        include: ["list"],
        embed: true,
      }),
    },
    models: {
      // userCurrent: Model,
      // user: Model,
      user: Model,
      company: Model,
      operator: Model,
      group: Model,
      forklift: Model,
      place: Model,
    },
    seeds(server) {
      // server.db.loadData({ userCurrent: isEmpty ? [] : userCurrentMock });
      server.db.loadData({ users: isEmpty ? [] : listUserMock });
      server.db.loadData({ groups: isEmpty ? [] : listGroupMock });
      server.db.loadData({ companies: isEmpty ? [] : listCompanyMock });
      server.db.loadData({ operators: isEmpty ? [] : listOperatorMock });
      server.db.loadData({ forklifts: isEmpty ? [] : listForkliftMock });
      server.db.loadData({ places: isEmpty ? [] : listPlaceMock });
    },
    routes() {
      this.namespace = "api/v0";
      userRoutes(this);
      companyRoutes(this);
      operatorRoutes(this);
      groupRoutes(this);
      forkliftRoutes(this);
      placeRoutes(this);
      this.passthrough("https://viacep.com.br/ws/**");
    },
  });
};
