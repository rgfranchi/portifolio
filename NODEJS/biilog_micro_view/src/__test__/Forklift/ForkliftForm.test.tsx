import React from "react";

import { act } from "react-dom/test-utils";

import { render, fireEvent, screen, cleanup } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { createMemoryHistory } from "history";

import store from "../../redux/store";

import { renderProviderRouter } from "../testHelper/test-provider";

import { defaultFormTextValidate } from "../testHelper/test-text-defaults";

import {
  forkliftMock,
  loadForkliftInvalidValues,
  loadForkliftValidValues,
} from "../server/serverForklift";

import ForkliftForm from "../../Forklift/ForkliftForm";
import { emptyForklift } from "../../Forklift/ForkliftData";
import { loadOperatorsByCompany } from "../server/serverOperator";
import { IOperator } from "../../Operator/OperatorData";
import { IPlace } from "../../Place/PlaceData";
import { loadPlacesByCompany } from "../server/serverPlace";

let wrapper: any = null;
let history: any = null;
const objectMock = forkliftMock;
const objectEmpty = emptyForklift;
let loadValues: { operators: IOperator[]; places: IPlace[] } = {
  operators: [],
  places: [],
};
const submitMock = jest.fn();

beforeEach(() => {
  cleanup();
  jest.setTimeout(20000);
  history = createMemoryHistory();
  loadValues.operators = loadOperatorsByCompany(objectMock.module.companyId);
  loadValues.places = loadPlacesByCompany(objectMock.module.companyId);
});

describe("ForkliftForm snapshot", () => {
  test("empty", async () => {
    history.entries[0].key = "EMPTY_01";
    await setWrapperDefault(objectEmpty, submitMock, loadValues);
    expect(wrapper.container).toMatchSnapshot();
  });
  test("with data", async () => {
    history.entries[0].key = "DATA_01";
    await setWrapperDefault(objectMock, submitMock, loadValues);
    expect(wrapper.container).toMatchSnapshot();
  });
});

describe("ForkliftForm validate", () => {
  test("valid", async () => {
    await setWrapperDefault(objectEmpty, submitMock, loadValues);
    await act(async () => {
      await loadForkliftValidValues(userEvent, fireEvent, screen);
    });
    expect(
      screen.queryByText(defaultFormTextValidate.preenchimento)
    ).toBeNull();
    await act(async () => {
      fireEvent.click(screen.getByText("Salvar"));
    });
    expect(submitMock).toHaveBeenCalledTimes(1);
  });
  test("invalid", async () => {
    await setWrapperDefault(objectEmpty, submitMock, loadValues);
    await act(async () => {
      fireEvent.click(screen.getByText("Salvar"));
    });
    expect(
      screen.queryAllByText(defaultFormTextValidate.preenchimento).length
    ).toBe(3);
    await act(async () => {
      await loadForkliftInvalidValues(userEvent, fireEvent, screen);
    });
    expect(
      screen.queryAllByText(defaultFormTextValidate.preenchimento).length
    ).toBe(2);
    expect(submitMock).toHaveBeenCalledTimes(1);
  });
});

const setWrapperDefault = async (
  object: any,
  handleSubmit: any = () => {},
  loadValues: any
) => {
  await act(async () => {
    wrapper = await renderProviderRouter(
      render,
      <ForkliftForm
        initValues={object}
        submitForm={handleSubmit}
        loadValues={loadValues}
      />,
      history,
      store
    );
  });
};
