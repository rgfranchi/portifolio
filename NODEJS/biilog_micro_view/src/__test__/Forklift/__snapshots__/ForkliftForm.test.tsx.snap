// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForkliftForm snapshot empty 1`] = `
<div>
  <form
    class=""
  >
    <div
      class="accordion"
    >
      <div
        class="card-header"
      >
        Equipamento
      </div>
      <div
        class="collapse show"
      >
        <div
          class="card-body"
        >
          <div
            class="form-row"
          >
            <div
              class="form-group col-md-4"
            >
              <label
                class="form-label"
                for="ctrlCodigo"
              >
                Código
              </label>
              <input
                class="form-control"
                id="ctrlCodigo"
                name="codigo"
                placeholder="Código do equipamento"
                type="text"
                value=""
              />
              <div
                class="invalid-feedback"
              />
            </div>
            <div
              class="form-group col-md-4"
            >
              <label
                class="form-label"
                for="ctrlFabricante"
              >
                Fabricante
              </label>
              <input
                class="form-control"
                id="ctrlFabricante"
                name="fabricante"
                placeholder="Fabricante"
                type="text"
                value=""
              />
            </div>
            <div
              class="form-group col-md-4"
            >
              <label
                class="form-label"
                for="ctrlModelo"
              >
                Modelo
              </label>
              <input
                class="form-control"
                id="ctrlModelo"
                name="modelo"
                placeholder="Modelo"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="form-row"
          >
            <div
              class="form-group col-md-3"
            >
              <label
                class="form-label"
                for="ctrlMotor"
              >
                Motor
              </label>
              <input
                class="form-control"
                id="ctrlMotor"
                name="motor"
                placeholder="Motor"
                type="text"
                value=""
              />
            </div>
            <div
              class="form-group col-md-3"
            >
              <label
                class="form-label"
                for="ctrlAnoFabricacao"
              >
                Ano de Fabricação
              </label>
              <input
                class="form-control"
                id="ctrlAnoFabricacao"
                name="anoFabricacao"
                placeholder="Ano"
                type="text"
                value=""
              />
            </div>
            <div
              class="form-group col-md-3"
            >
              <label
                class="form-label"
                for="ctrlHorimetroInicial"
              >
                Horímetro inicial
              </label>
              <input
                class="form-control"
                id="ctrlHorimetroInicial"
                name="horimetroInicial"
                placeholder="Horímetro"
                type="text"
                value="0"
              />
            </div>
            <div
              class="form-group col-md-3"
            >
              <label
                class="form-label"
                for="ctrlHodometroInicial"
              >
                Hodômetro inicial
              </label>
              <input
                class="form-control"
                id="ctrlHodometroInicial"
                name="hodometroInicial"
                placeholder="Hodômetro"
                type="text"
                value="0"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="accordion"
    >
      <div
        class="card-header"
      >
        Operadores
      </div>
      <div
        class="collapse show"
      >
        <div
          class="card-body"
        >
          <div
            class="row"
          >
            <div
              class="col"
            >
              <button
                class="btn btn-primary btn-block"
                id="1"
                type="button"
              >
                Thiago de Lima Esmero
              </button>
            </div>
            <div
              class="col"
            >
              <button
                class="btn btn-primary btn-block"
                id="2"
                type="button"
              >
                JN Manutenções
              </button>
            </div>
            <div
              class="col"
            >
              <button
                class="btn btn-primary btn-block"
                id="7"
                type="button"
              >
                Operador padrão WINE
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="accordion"
    >
      <div
        class="card-header"
      >
        Módulo
      </div>
      <div
        class="collapse show"
      >
        <div
          class="card-body"
        >
          <div
            class="form-row"
          >
            <div
              class="form-group col-md-3"
            >
              <label
                class="form-label"
                for="ctrlModule_Codigo"
              >
                Código Modulo
              </label>
              <input
                class="form-control"
                id="ctrlModule_Codigo"
                name="module.codigo"
                placeholder="Código do módulo"
                type="text"
                value=""
              />
              <div
                class="invalid-feedback"
              />
            </div>
            <div
              class="form-group col-md-1"
            >
              <label
                class="form-label"
                for="ctrlModule_ativo"
              >
                Ativo
              </label>
              <button
                class="btn btn-success form-control"
                id="ctrlModule_ativo"
                name="module.ativo"
              >
                Ativo
              </button>
            </div>
            <div
              class="form-group col-md-3"
            >
              <label
                class="form-label"
                for="ctrlModule_modelo"
              >
                Modelo
              </label>
              <input
                class="form-control"
                id="ctrlModule_modelo"
                name="module.modelo"
                placeholder="Modelo Modulo"
                type="text"
                value=""
              />
            </div>
            <div
              class="form-group col-md-3"
            >
              <label
                class="form-label"
                for="ctrlModule_fabricante"
              >
                Fabricante
              </label>
              <input
                class="form-control"
                id="ctrlModule_fabricante"
                name="module.fabricante"
                placeholder="Fabricante do módulo"
                type="text"
                value=""
              />
            </div>
            <div
              class="form-group col-md-2"
            >
              <label
                class="form-label"
                for="ctrlModule_Place"
              >
                Localização
              </label>
              <select
                class="form-control"
                id="ctrlModule_Place"
                name="module.placeId"
              >
                <option
                  value="2"
                />
              </select>
            </div>
          </div>
          <div
            class="form-row"
          />
          <div
            class="form-row"
          >
            <div
              class="form-group col"
            >
              <label
                class="form-label"
                for="ctrlModule_observacoes"
              >
                Observações
              </label>
              <textarea
                class="form-control"
                id="ctrlModule_observacoes"
                name="module.observacoes"
                placeholder="Informações do módulo ..."
                rows="4"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="accordion"
    >
      <div
        class="card-header"
      >
        Transmissão
      </div>
      <div
        class="collapse show"
      >
        <div
          class="card-body"
        >
          <div
            class="form-row"
          >
            <div
              class="form-group col-md-4"
            >
              <label
                class="form-label"
                for="ctrlTransmission_Tipo"
              >
                Tipo Transmissão
              </label>
              <input
                class="form-control"
                id="ctrlTransmission_Tipo"
                name="module.transmission.tipo"
                placeholder="Tipo de transmissão"
                type="text"
                value=""
              />
              <div
                class="invalid-feedback"
              />
            </div>
            <div
              class="form-group col-md-4"
            >
              <label
                class="form-label"
                for="ctrlTransmission_Fabricante"
              >
                Fabricante
              </label>
              <input
                class="form-control"
                id="ctrlTransmission_Fabricante"
                name="module.transmission.fabricante"
                placeholder="Fabricante da Transmissão"
                type="text"
                value=""
              />
            </div>
            <div
              class="form-group col-md-4"
            >
              <label
                class="form-label"
                for="ctrlTransmission_Modelo"
              >
                Modelo
              </label>
              <input
                class="form-control"
                id="ctrlTransmission_Modelo"
                name="module.transmission.modelo"
                placeholder="Modelo Transmissão"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="form-row"
          >
            <div
              class="form-group col-md-3"
            >
              <label
                class="form-label"
                for="ctrlTransmission_Chip"
              >
                Chip
              </label>
              <input
                class="form-control"
                id="ctrlTransmission_Chip"
                name="module.transmission.chip"
                placeholder="Chip"
                type="text"
                value=""
              />
            </div>
            <div
              class="form-group col-md-3"
            >
              <label
                class="form-label"
                for="ctrlTransmission_Operadora"
              >
                Operadora
              </label>
              <input
                class="form-control"
                id="ctrlTransmission_Operadora"
                name="module.transmission.operadora"
                placeholder="Operadora"
                type="text"
                value=""
              />
            </div>
            <div
              class="form-group col-md-3"
            >
              <label
                class="form-label"
                for="ctrlTransmission_NumeroLinha"
              >
                Número Linha
              </label>
              <input
                class="form-control"
                id="ctrlTransmission_NumeroLinha"
                name="module.transmission.numeroLinha"
                placeholder="(99)9999-9999"
                type="text"
                value=""
              />
            </div>
            <div
              class="form-group col-md-3"
            >
              <label
                class="form-label"
                for="ctrlTransmission_Pin"
              >
                PIN
              </label>
              <input
                class="form-control"
                id="ctrlTransmission_Pin"
                name="module.transmission.pin"
                placeholder="9999"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="form-row"
          >
            <div
              class="form-group col-md-2"
            >
              <label
                class="form-label"
                for="ctrlTransmission_Puk1"
              >
                PUK 1
              </label>
              <input
                class="form-control"
                id="ctrlTransmission_Puk1"
                name="module.transmission.puk1"
                placeholder="99999999"
                type="text"
                value=""
              />
            </div>
            <div
              class="form-group col-md-2"
            >
              <label
                class="form-label"
                for="ctrlTransmission_Puk2"
              >
                PUK 2
              </label>
              <input
                class="form-control"
                id="ctrlTransmission_Puk2"
                name="module.transmission.puk2"
                placeholder="99999999"
                type="text"
                value=""
              />
            </div>
            <div
              class="form-group col-md-2"
            >
              <label
                class="form-label"
                for="ctrlTransmission_Puk3"
              >
                PUK 3
              </label>
              <input
                class="form-control"
                id="ctrlTransmission_Puk3"
                name="module.transmission.puk3"
                placeholder="99999999"
                type="text"
                value=""
              />
            </div>
            <div
              class="form-group col-md-3"
            >
              <label
                class="form-label"
                for="ctrlTransmission_IMEI"
              >
                IMEI
              </label>
              <input
                class="form-control"
                id="ctrlTransmission_IMEI"
                name="module.transmission.imei"
                placeholder="99999999"
                type="text"
                value=""
              />
            </div>
            <div
              class="form-group col-md-3"
            >
              <label
                class="form-label"
                for="ctrlTransmission_PAN"
              >
                PAN
              </label>
              <input
                class="form-control"
                id="ctrlTransmission_PAN"
                name="module.transmission.pan"
                placeholder="99999999"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="form-row"
          >
            <div
              class="form-group col"
            >
              <label
                class="form-label"
                for="ctrlTransmission_observacoes"
              >
                Observações
              </label>
              <textarea
                class="form-control"
                id="ctrlTransmission_observacoes"
                name="module.transmission.observacoes"
                placeholder="Informações da transmissão ..."
                rows="4"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      class="btn btn-primary"
      type="submit"
    >
      Salvar
    </button>
  </form>
</div>
`;

exports[`ForkliftForm snapshot with data 1`] = `
<div>
  <form
    class=""
  >
    <div
      class="accordion"
    >
      <div
        class="card-header"
      >
        Equipamento
      </div>
      <div
        class="collapse show"
      >
        <div
          class="card-body"
        >
          <div
            class="form-row"
          >
            <div
              class="form-group col-md-4"
            >
              <label
                class="form-label"
                for="ctrlCodigo"
              >
                Código
              </label>
              <input
                class="form-control"
                id="ctrlCodigo"
                name="codigo"
                placeholder="Código do equipamento"
                type="text"
                value="ES-18 NEW"
              />
              <div
                class="invalid-feedback"
              />
            </div>
            <div
              class="form-group col-md-4"
            >
              <label
                class="form-label"
                for="ctrlFabricante"
              >
                Fabricante
              </label>
              <input
                class="form-control"
                id="ctrlFabricante"
                name="fabricante"
                placeholder="Fabricante"
                type="text"
                value="Hyster NEW"
              />
            </div>
            <div
              class="form-group col-md-4"
            >
              <label
                class="form-label"
                for="ctrlModelo"
              >
                Modelo
              </label>
              <input
                class="form-control"
                id="ctrlModelo"
                name="modelo"
                placeholder="Modelo"
                type="text"
                value="R30XMA3"
              />
            </div>
          </div>
          <div
            class="form-row"
          >
            <div
              class="form-group col-md-3"
            >
              <label
                class="form-label"
                for="ctrlMotor"
              >
                Motor
              </label>
              <input
                class="form-control"
                id="ctrlMotor"
                name="motor"
                placeholder="Motor"
                type="text"
                value="Elétrica 36v"
              />
            </div>
            <div
              class="form-group col-md-3"
            >
              <label
                class="form-label"
                for="ctrlAnoFabricacao"
              >
                Ano de Fabricação
              </label>
              <input
                class="form-control"
                id="ctrlAnoFabricacao"
                name="anoFabricacao"
                placeholder="Ano"
                type="text"
                value="2099"
              />
            </div>
            <div
              class="form-group col-md-3"
            >
              <label
                class="form-label"
                for="ctrlHorimetroInicial"
              >
                Horímetro inicial
              </label>
              <input
                class="form-control"
                id="ctrlHorimetroInicial"
                name="horimetroInicial"
                placeholder="Horímetro"
                type="text"
                value="77777"
              />
            </div>
            <div
              class="form-group col-md-3"
            >
              <label
                class="form-label"
                for="ctrlHodometroInicial"
              >
                Hodômetro inicial
              </label>
              <input
                class="form-control"
                id="ctrlHodometroInicial"
                name="hodometroInicial"
                placeholder="Hodômetro"
                type="text"
                value="77777"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="accordion"
    >
      <div
        class="card-header"
      >
        Operadores
      </div>
      <div
        class="collapse show"
      >
        <div
          class="card-body"
        >
          <div
            class="row"
          >
            <div
              class="col"
            >
              <button
                class="btn btn-primary btn-block"
                id="1"
                type="button"
              >
                Thiago de Lima Esmero
              </button>
            </div>
            <div
              class="col"
            >
              <button
                class="btn btn-primary btn-block"
                id="2"
                type="button"
              >
                JN Manutenções
              </button>
            </div>
            <div
              class="col"
            >
              <button
                class="btn btn-primary btn-block"
                id="7"
                type="button"
              >
                Operador padrão WINE
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="accordion"
    >
      <div
        class="card-header"
      >
        Módulo
      </div>
      <div
        class="collapse show"
      >
        <div
          class="card-body"
        >
          <div
            class="form-row"
          >
            <div
              class="form-group col-md-3"
            >
              <label
                class="form-label"
                for="ctrlModule_Codigo"
              >
                Código Modulo
              </label>
              <input
                class="form-control"
                id="ctrlModule_Codigo"
                name="module.codigo"
                placeholder="Código do módulo"
                type="text"
                value="99ABCNEW"
              />
              <div
                class="invalid-feedback"
              />
            </div>
            <div
              class="form-group col-md-1"
            >
              <label
                class="form-label"
                for="ctrlModule_ativo"
              >
                Ativo
              </label>
              <button
                class="btn btn-success form-control"
                id="ctrlModule_ativo"
                name="module.ativo"
              >
                Ativo
              </button>
            </div>
            <div
              class="form-group col-md-3"
            >
              <label
                class="form-label"
                for="ctrlModule_modelo"
              >
                Modelo
              </label>
              <input
                class="form-control"
                id="ctrlModule_modelo"
                name="module.modelo"
                placeholder="Modelo Modulo"
                type="text"
                value="ELÉTRICO New"
              />
            </div>
            <div
              class="form-group col-md-3"
            >
              <label
                class="form-label"
                for="ctrlModule_fabricante"
              >
                Fabricante
              </label>
              <input
                class="form-control"
                id="ctrlModule_fabricante"
                name="module.fabricante"
                placeholder="Fabricante do módulo"
                type="text"
                value="BIILOG"
              />
            </div>
            <div
              class="form-group col-md-2"
            >
              <label
                class="form-label"
                for="ctrlModule_Place"
              >
                Localização
              </label>
              <select
                class="form-control"
                id="ctrlModule_Place"
                name="module.placeId"
              >
                <option
                  value="2"
                />
              </select>
            </div>
          </div>
          <div
            class="form-row"
          />
          <div
            class="form-row"
          >
            <div
              class="form-group col"
            >
              <label
                class="form-label"
                for="ctrlModule_observacoes"
              >
                Observações
              </label>
              <textarea
                class="form-control"
                id="ctrlModule_observacoes"
                name="module.observacoes"
                placeholder="Informações do módulo ..."
                rows="4"
              >
                CNHi Modulo MOCK NEW
              </textarea>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="accordion"
    >
      <div
        class="card-header"
      >
        Transmissão
      </div>
      <div
        class="collapse show"
      >
        <div
          class="card-body"
        >
          <div
            class="form-row"
          >
            <div
              class="form-group col-md-4"
            >
              <label
                class="form-label"
                for="ctrlTransmission_Tipo"
              >
                Tipo Transmissão
              </label>
              <input
                class="form-control"
                id="ctrlTransmission_Tipo"
                name="module.transmission.tipo"
                placeholder="Tipo de transmissão"
                type="text"
                value="Zigbee"
              />
              <div
                class="invalid-feedback"
              />
            </div>
            <div
              class="form-group col-md-4"
            >
              <label
                class="form-label"
                for="ctrlTransmission_Fabricante"
              >
                Fabricante
              </label>
              <input
                class="form-control"
                id="ctrlTransmission_Fabricante"
                name="module.transmission.fabricante"
                placeholder="Fabricante da Transmissão"
                type="text"
                value="NEW Biilog"
              />
            </div>
            <div
              class="form-group col-md-4"
            >
              <label
                class="form-label"
                for="ctrlTransmission_Modelo"
              >
                Modelo
              </label>
              <input
                class="form-control"
                id="ctrlTransmission_Modelo"
                name="module.transmission.modelo"
                placeholder="Modelo Transmissão"
                type="text"
                value="GATEWAY NEW"
              />
            </div>
          </div>
          <div
            class="form-row"
          >
            <div
              class="form-group col-md-3"
            >
              <label
                class="form-label"
                for="ctrlTransmission_Chip"
              >
                Chip
              </label>
              <input
                class="form-control"
                id="ctrlTransmission_Chip"
                name="module.transmission.chip"
                placeholder="Chip"
                type="text"
                value="256325422554"
              />
            </div>
            <div
              class="form-group col-md-3"
            >
              <label
                class="form-label"
                for="ctrlTransmission_Operadora"
              >
                Operadora
              </label>
              <input
                class="form-control"
                id="ctrlTransmission_Operadora"
                name="module.transmission.operadora"
                placeholder="Operadora"
                type="text"
                value="TIM"
              />
            </div>
            <div
              class="form-group col-md-3"
            >
              <label
                class="form-label"
                for="ctrlTransmission_NumeroLinha"
              >
                Número Linha
              </label>
              <input
                class="form-control"
                id="ctrlTransmission_NumeroLinha"
                name="module.transmission.numeroLinha"
                placeholder="(99)9999-9999"
                type="text"
                value="(11)40046895"
              />
            </div>
            <div
              class="form-group col-md-3"
            >
              <label
                class="form-label"
                for="ctrlTransmission_Pin"
              >
                PIN
              </label>
              <input
                class="form-control"
                id="ctrlTransmission_Pin"
                name="module.transmission.pin"
                placeholder="9999"
                type="text"
                value="3698"
              />
            </div>
          </div>
          <div
            class="form-row"
          >
            <div
              class="form-group col-md-2"
            >
              <label
                class="form-label"
                for="ctrlTransmission_Puk1"
              >
                PUK 1
              </label>
              <input
                class="form-control"
                id="ctrlTransmission_Puk1"
                name="module.transmission.puk1"
                placeholder="99999999"
                type="text"
                value="3677"
              />
            </div>
            <div
              class="form-group col-md-2"
            >
              <label
                class="form-label"
                for="ctrlTransmission_Puk2"
              >
                PUK 2
              </label>
              <input
                class="form-control"
                id="ctrlTransmission_Puk2"
                name="module.transmission.puk2"
                placeholder="99999999"
                type="text"
                value="3666"
              />
            </div>
            <div
              class="form-group col-md-2"
            >
              <label
                class="form-label"
                for="ctrlTransmission_Puk3"
              >
                PUK 3
              </label>
              <input
                class="form-control"
                id="ctrlTransmission_Puk3"
                name="module.transmission.puk3"
                placeholder="99999999"
                type="text"
                value="3655"
              />
            </div>
            <div
              class="form-group col-md-3"
            >
              <label
                class="form-label"
                for="ctrlTransmission_IMEI"
              >
                IMEI
              </label>
              <input
                class="form-control"
                id="ctrlTransmission_IMEI"
                name="module.transmission.imei"
                placeholder="99999999"
                type="text"
                value="3132165164554635"
              />
            </div>
            <div
              class="form-group col-md-3"
            >
              <label
                class="form-label"
                for="ctrlTransmission_PAN"
              >
                PAN
              </label>
              <input
                class="form-control"
                id="ctrlTransmission_PAN"
                name="module.transmission.pan"
                placeholder="99999999"
                type="text"
                value="AJSPT3216549"
              />
            </div>
          </div>
          <div
            class="form-row"
          >
            <div
              class="form-group col"
            >
              <label
                class="form-label"
                for="ctrlTransmission_observacoes"
              >
                Observações
              </label>
              <textarea
                class="form-control"
                id="ctrlTransmission_observacoes"
                name="module.transmission.observacoes"
                placeholder="Informações da transmissão ..."
                rows="4"
              >
                Transmissão do MOCK NEW
              </textarea>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      class="btn btn-primary"
      type="submit"
    >
      Salvar
    </button>
  </form>
</div>
`;
